ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -267,6 +322,29 @@ private CloudBlobContainer getContainer(BlobId blobId, boolean autoCreate)
     return azureContainer;
   }
 
+  /**
+   * Get the blob name to use in Azure Blob Storage
+   * @param blobId The {@link BlobId} to store.
+   * @return An Azure-friendly blob name.
+   */
+  private String getAzureBlobName(BlobId blobId) {
+    // Prefix to assist in blob data sharding, since beginning of blobId has little variation.
+    String prefix = blobId.getUuid().substring(0, 4) + "-";
+    return prefix + blobId.getID();
+  }
+
+  /**
+   * Update the appropriate error metrics corresponding to the thrown exception.
+   * @param e the exception thrown.
+   */
+  private void updateErrorMetrics(Exception e) {
+    if (e instanceof DocumentClientException) {
+      azureMetrics.documentErrorCount.inc();
+    } else {
```

Comment:
```
else if (e instanceof StorageException) {
}
```
To exclude  `URISyntaxException` explicitly.

Reply:
azureContainer.getBlockBlobReference() can throw URISyntaxException, and this should count as a storageError. (Though unlikely to happen in practice.)